web/src/engine/osk/src/input/gestures/browser/subkeyPopup.ts

```diff
@@ -112,20 +117,100 @@ export default class SubkeyPopup implements RealizedGesture {
 
     // Highlight the duplicated base key or ideal subkey (if a phone)
     if(vkbd.device.formFactor == DeviceSpec.FormFactor.Phone) {
-      this.selectDefaultSubkey(vkbd, e, subKeys /* == this.element */);
+      this.selectDefaultSubkey(e, elements /* == this.element */);
     }
+
+    vkbd.topContainer.appendChild(this.element);
+    vkbd.topContainer.appendChild(this.shim);
+
+    // Must be placed after its `.element` has been inserted into the DOM.
+    this.reposition(vkbd);
+
+    const config = this.buildPopupRecognitionConfig(vkbd);
+    configChanger({
+      type: 'push',
+      config: config
+    });
   }
 
-  finalize(input: InputSample<KeyElement>) {
-    if(this.resolver) {
-      let keyEvent: KeyEvent = null;
-      if(this.currentSelection) {
-        keyEvent = this.vkbd.initKeyEvent(this.currentSelection, input);
-        this.currentSelection.key.highlight(false);
+  private buildPopupRecognitionConfig(vkbd: VisualKeyboard): GestureRecognizerConfiguration<KeyElement, string> {
+    const baseBounding = this.element.getBoundingClientRect();
+    const underlyingKeyBounding = this.baseKey.getBoundingClientRect();
+
+    const subkeyStyle = this.subkeys[0].style;
+    const subkeyHeight = Number.parseInt(subkeyStyle.height, 10);
+    const basePadding = -0.666 * subkeyHeight;  // extends bounds by the absolute value.
+
+    const bottomDistance = underlyingKeyBounding.bottom - baseBounding.bottom;
+
+    const roamBounding = new PaddedZoneSource(this.element, [
+      // top
+      basePadding,
+      // left, right
+      basePadding,
+      // bottom: ensure the recognition zone includes the row of the base key.
+      // basePadding is already negative, but bottomDistance isn't.
+      -bottomDistance < basePadding ? -bottomDistance : basePadding
+    ]);
```

Comment:
This is the section that defines the 'roaming bounds' that enable and disable key selection.  Note the `basePadding` entry - that value in particular is what defines the maximum overshoot we're allowing.

Note:  values here are negative for 'external padding' - essentially, CSS 'margin', but positive for 'internal padding' - essentially, CSS 'padding'.